src/offset.js

```diff
@@ -92,7 +92,12 @@ jQuery.fn.extend({
 
 		rect = elem.getBoundingClientRect();
 
-		// Make sure element is not hidden (display: none) or disconnected
+		// Return zeros for disconnected elements (gh-2310)
+		if ( !jQuery.contains( document, elem ) ) {
```

Comment:
I tried that first, but I'm trying to separate hidden elements from disconnected ones. Should we return zeros for hidden elements too?